tests/test_flow/test_persistence_random.py

```diff
@@ -200,7 +200,7 @@ def _add_binding_to_flow(self, binding):
             output_value_fragment = binding.value_code_fragment_for_outputs()
 
         vars_dict = {
-            "bn": bn,
+            **globals(),
```

Comment:
This is directly related to one of your [previous comment](https://github.com/square/bionic/pull/299#discussion_r530565556). I didn't know that functions can have a `None` module. But I uncovered that from this test. If we don't pass it `**globals()` here, then the function that we define here does not have a module associated with it.

If you think adding all globals is excessive, I can use something like `import_code` helper method for this instead. Wdyt?

Reply:
I think I'm missing two things here:

1. If we just want this function to have a `__module__`, why do we need to include all of `globals()`? Is there a particular global variable that determines what the module is?

2. Why do we choose not to scan functions that have empty modules?

Reply:
1. Right, passing `globals()` also passes some variable that defines the module for the exec code.

2. Well, we decided to not hash any variables that are either in `Bionic`, or is an internal module (anything that lives in `sysconfig_paths()`, which is what our reload functionality does too).
I don't have a good reason to include or not include coroutines without modules. There also isn't enough information out there that tells us when `__module__` object isn't available. So I decided to ignore these objects since they can have unintended side effects. But I can be easily convinced otherwise.

Reply:
1. So could we just pass in that particular variable?

2. What side effects did you have in mind? I don't have a good sense either about whether to include these or not, but I would probably err on the side of hashing them. (My intuition is that these functions are more likely to be user functions than library functions, but I could be wrong.) Either way, let's leave a comment with our rationale.

Reply:
On second thought, we only check module to see if it's a Bionic variable, and any Bionic variable should have that variable. So I'll hash variables if thy don't have a variable name.